### PR TITLE
tests: get mender logs as part of html report

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Changelog: none
Signed-off-by: Greg Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

File: tests/helpers.py [hunk 1]
```diff
@@ -207,3 +207,20 @@ def ip_to_device_id_map(clients):
...
+    @staticmethod
+    def get_mender_logs():
+        output_from_deployment_logs = ""
+
+        try:
+            output_from_journalctl = run("journalctl -u mender -l")
+            logs_exist = int(run("find '/data/mender' -name '*.log' -type f | wc -l").strip())
+
+            if logs_exist:
+                output_from_deployment_logs = run("grep '' -H /data/mender/*.log")
```
Comment (kacf) [inline tests/helpers.py]: Grep for nothing to get the name. Clever!

File: tests/helpers.py [hunk 2]
```diff
@@ -207,3 +207,20 @@ def ip_to_device_id_map(clients):
...
+
+        except (BaseException):
+            logger.warn("failed to grab logs from mender client")
+            return None, None
+        else:
+            return output_from_journalctl, output_from_deployment_logs
```
Comment (kacf) [inline tests/helpers.py]: `output_from_journalctl` is defined inside the `try`. Do you need one outside to make sure it stays in scope? Like with `output_from_deployment_logs`.

File: tests/helpers.py [hunk 3]
```diff
@@ -207,3 +207,20 @@ def ip_to_device_id_map(clients):
...
 
         return ip_to_device_id
+
+    @staticmethod
+    def get_mender_logs():
+        output_from_deployment_logs = ""
+
+        try:
+            output_from_journalctl = run("journalctl -u mender -l")
+            logs_exist = int(run("find '/data/mender' -name '*.log' -type f | wc -l").strip())
```
Comment (bboozzoo) [inline tests/helpers.py]: Isn't the deployment log removed once it has been uploaded to the backend?

File: tests/helpers.py (hunk 2 above)
Comment (bboozzoo) [reply tests/helpers.py]: That's not how scope works in Python

Comment (bboozzoo) [reply tests/helpers.py]: That's not how scope works in Python

**Comment by bboozzoo** [comment]
btw. needs a rebase

**Comment by kacf** [comment]
Outdated.

**Pull request closed without merging**